### PR TITLE
Document state machine

### LINKED ISSUE #22: Document network state machine in README
E.g. as ASCII art state machine. It's important that all states and transitions are visible.

Alternatively we could write it in dot syntax and generate a PNG.

### PR DESCRIPTION
**Pull request by dbrgn**
Fixes #22.

@rnestler, is this correct?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,31 @@ If everything built fine you can use cargo to run the client or the server:
...
+                 | | +----------------+   +---------------------+ | |
+                 | +->Command(Command)+--->UpdateGamestate(Game)+-+ |
+                 |   +----------------+   +---------------------+   |
+                 |                                                  |
+                 |   +-------------------------+                    |
+                 +--->ClientReconnect(ClientId)+--------------------+
+                     +-------------------------+
+
+- Initially, the server waits for a `ClientHello` message.
+- It responds with a `ServerHello` message that contains the client ID that can
```
Comment (rnestler) [inline README.md:76]: The server also creates the new player at this point.

**Comment by rnestler** [comment]
I noticed that we should handle and document the case when a client want's to reconnect to an already connected client ID. I'm not sure what currently would happen...

File: README.md (hunk 1 above)
Comment (dbrgn) [inline README.md:76]: Yes, but that's an implementation detail.

**Comment by rnestler** [comment]
@dbrgn I noticed that its pretty wrong actually: As soon as a client connects, the server starts sending updates in a loop. Updates happen all the time because of other players and since time is moving along.

**Pull request merged**